src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -306,6 +335,101 @@ .maxstack  4
 }
 ");
         }
+        [Fact]
+        public void TupleTypeDeclaration()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        (int, string, int) x = (1, ""hello"", 2);
+        System.Console.WriteLine(x.ToString());
+    }
+}
+
+" + trivial3uple;
+
+            var comp = CompileAndVerify(source, expectedOutput: @"{1, hello, 2}");
+        }
+
+        [Fact]
+        public void TupleTypeMismatch()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        (int, string) x = (1, ""hello"", 2);
+    }
+}
+" + trivial2uple + trivial3uple;
+
+            // PROTOTYPE there should be no mention of Item1, Item2 or Item3 in there error for this test
```

Comment:
What I'm thinking is that `(int a, string b)` should print as `<tuple: int a, string b>` and `(int, string)` should print out as `<tuple: int, string>` (not `<tuple:int Item1, string Item2>`).
Only user-provided names should appear in the TupleTypeSymbol. The underlying type (`ValueTuple`) is the one that provides Item1, etc. 

At the moment, because our error checking is incomplete, and also for purpose of error recovery, we assign "Item1" and such into the tuple symbol when the user missed some names. But I think this is a prototype bug.